src/main/java/pl/edu/icm/pl/mxrdr/extension/workflow/utils/FileContentReplacer.java

```diff
@@ -0,0 +1,93 @@
+package pl.edu.icm.pl.mxrdr.extension.workflow.utils;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class FileContentReplacer {
+    private final String fileName;
+    private final Path workDir;
+    private Set<PatternAndReplacement> actions = new HashSet<>();
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    private FileContentReplacer(String fileName, Path workDir) {
+        this.fileName = fileName;
+        this.workDir = workDir;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public static FileContentReplacer of(String fileName, Path workDir) {
+        return new FileContentReplacer(fileName, workDir);
+    }
+
+    public FileContentReplacer add(PatternAndReplacement action) {
+        if (action != PatternAndReplacement.NONE) {
```

Comment:
Risky business comparing regular objects by their references. Maybe better just check for empty pattern?

Reply:
This one is intentional as `PatternAndReplacement.NONE` is a kind of null object for that class, so it should only filter out cases where that object is explicitly used.

Reply:
I know it's intentional. I'm saying it's unsafe. Nothing stops me from creating another instance of `PatternAndReplacement` with the same filed values as `NONE` and this would not be a valid object. Hence limiting the condition to this single instance only can lead to errors in runtime. Extending the condition to instead check the emptiness of value of the `pattern` property would prevent such potential errors.